core/src/main/java/org/elasticsearch/action/search/SearchDfsQueryAndFetchAsyncAction.java

```diff
@@ -46,9 +47,10 @@
     SearchDfsQueryAndFetchAsyncAction(ESLogger logger, SearchTransportService searchTransportService,
                                               ClusterService clusterService, IndexNameExpressionResolver indexNameExpressionResolver,
                                               SearchPhaseController searchPhaseController, ThreadPool threadPool,
-                                              SearchRequest request, ActionListener<SearchResponse> listener) {
+                                              SearchRequest request, ActionListener<SearchResponse> listener,
+                                              PluginsService pluginsService) {
```

Comment:
You might even be able to get away with not making the interface and using `BiConsumer`.